File: MediTrack/notification-service/src/main/java/com/meditrack/notification/service/NotificationService.java

package com.meditrack.notification.service;

import com.meditrack.notification.entity.Notification;
import com.meditrack.notification.repository.NotificationRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class NotificationService {

    private final NotificationRepository repo;

    public NotificationService(
            NotificationRepository repo
    ) {
        this.repo = repo;
    }

    // CREATE
    public Notification create(
            Notification n
    ) {
        return repo.save(n);
    }

    // PATIENT LIST
    public List<Notification> getByPatient(
            Long patientId
    ) {
        return repo
          .findByPatientIdOrderByCreatedAtDesc(
                  patientId
          );
    }

    // DOCTOR LIST 
    public List<Notification> getByDoctor(
            Long doctorId
    ) {
        return repo
          .findByDoctorIdOrderByCreatedAtDesc(
                  doctorId
          );
    }

    // PATIENT UNREAD
    public long unread(
            Long patientId
    ) {
        return repo
          .countByPatientIdAndIsReadFalse(
                  patientId
          );
    }

    // DOCTOR UNREAD 
    public long unreadDoctor(
            Long doctorId
    ) {
        return repo
          .countByDoctorIdAndIsReadFalse(
                  doctorId
          );
    }

    // MARK READ
    public void markRead(
            Long id
    ) {
        Notification n =
            repo.findById(id)
                .orElseThrow();

        n.setRead(true);

        repo.save(n);
    }
}

File: c_meditrack-frontend/src/api/notificationService.js
import axiosInstance from "./axiosInstance";

/* ===============================
   PATIENT NOTIFICATIONS
================================= */

// Get all patient notifications
export const getMyNotifications = (patientId) =>
  axiosInstance.get(`/notifications/patient/${patientId}`);

// Patient unread count
export const getUnreadCount = (patientId) =>
  axiosInstance.get(`/notifications/unread/${patientId}`);

/* ===============================
   DOCTOR NOTIFICATIONS
================================= */

// Get all doctor notifications
export const getDoctorNotifications = (doctorId) =>
  axiosInstance.get(`/notifications/doctor/${doctorId}`);

// Doctor unread count
export const getDoctorUnreadCount = (doctorId) =>
  axiosInstance.get(`/notifications/doctor/unread/${doctorId}`);

/* ===============================
   COMMON
================================= */

// Mark single notification as read
export const markNotificationRead = (id) =>
  axiosInstance.put(`/notifications/read/${id}`);